src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java

```diff
@@ -237,6 +240,33 @@ public Builder sdkMetricsTags(Map<String, String> tags) {
       return this;
     }
 
+  /**
+   * For a given server endpoint, validate according to RFC 2396 and attempt
+   * to make a connection
+   *
+   * @return {@code this}
+   * @throws IllegalStateException
+   */
+    public Builder validateEndpoint() throws IllegalStateException {
+      URL url = null;
+
+      try {
+        url = new URL(this.server);
+      } catch (MalformedURLException e) {
+        throw new IllegalStateException("Please provide a reachable Wavefront endpoint", e);
+      }
+
+      try {
+        HttpURLConnection urlConn = (HttpURLConnection) url.openConnection();
+        urlConn.connect();
+        urlConn.disconnect();
+      } catch (IOException e) {
+        throw new IllegalStateException(e);
```

Comment:
ditto

Reply:
And now _this_ is the place to provide details like `"Unable to connect to " + this.server`